pkg/cmd/operator/mustgather.go

```diff
@@ -0,0 +1,290 @@
+		Name:      "scylla-operator",
+	},
+	{
+		GroupResource: schema.GroupResource{
+			Resource: "namespaces",
+			Group:    "",
+		},
+		Namespace: corev1.NamespaceAll,
+		Name:      "scylla-manager",
+	},
+	{
+		GroupResource: schema.GroupResource{
+			Resource: "validatingwebhookconfigurations",
+			Group:    "admissionregistration.k8s.io",
+		},
+		Namespace: corev1.NamespaceAll,
+		Name:      "",
+	},
+	{
+		GroupResource: schema.GroupResource{
+			Resource: "mutatingwebhookconfigurations",
+			Group:    "admissionregistration.k8s.io",
+		},
+		Namespace: corev1.NamespaceAll,
+		Name:      "",
+	},
+}
+
+func (o *MustGatherOptions) run(ctx context.Context) error {
+	startTime := time.Now()
+	klog.InfoS("Gathering artifacts", "TargetDir", o.DestDir)
+	defer func() {
+		klog.InfoS("Finished gathering artifacts", "Duration", time.Since(startTime))
+	}()
+
+	collector := collect.NewCollector(
+		o.DestDir,
+		o.GetPrinters(),
+		o.discoveryClient,
+		o.kubeClient.CoreV1(),
+		o.dynamicClient,
+		true,
+		o.KeepGoing,
+	)
+
+	var resourceSpecs []resourceSpec
+	if o.AllResources {
+		allPreferredListableResources, err := collector.DiscoverResources(ctx, discovery.SupportsAllVerbs{
+			Verbs: []string{"list"},
+		}.Match)
+		if err != nil {
+			return err
+		}
+
+		// Filter out native resources that share storage across groups.
+		preferredListableResources, err := collect.ReplaceIsometricresourceInfosIfPresent(allPreferredListableResources)
+		if err != nil {
+			return fmt.Errorf("can't replace isometric resourceInfos: %w", err)
+		}
+
+		resourceSpecs = make([]resourceSpec, 0, len(preferredListableResources))
+		for _, pr := range preferredListableResources {
+			resourceSpecs = append(resourceSpecs, resourceSpec{
+				ResourceInfo: collect.ResourceInfo{
+					Resource: pr.Resource,
+					Scope:    pr.Scope,
+				},
+				Namespace: corev1.NamespaceAll,
+				Name:      "",
+			})
+		}
+	} else {
+		preferredResources, err := collector.DiscoverResources(ctx, func(groupVersion string, r *metav1.APIResource) bool {
+			return true
+		})
+		if err != nil {
+			return err
+		}
+
+		resourceSpecs = make([]resourceSpec, 0, len(mustGatherSpecs))
+		for _, s := range mustGatherSpecs {
+			ri, err := findResource(preferredResources, s.GroupResource)
+			if err != nil {
+				return err
+			}
+
+			resourceSpecs = append(resourceSpecs, resourceSpec{
+				ResourceInfo: *ri,
+				Namespace:    corev1.NamespaceAll,
+				Name:         "",
+			})
+		}
+	}
+
+	var err error
+	var errs []error
+	for _, rs := range resourceSpecs {
+		if len(rs.Name) != 0 {
+			err = collector.CollectResource(ctx, &rs.ResourceInfo, rs.Namespace, rs.Name)
+			if apierrors.IsNotFound(err) {
+				klog.InfoS("Resource not found", "Resource", rs.ResourceInfo.Resource)
+				err = nil
+			}
+		} else {
+			err = collector.CollectResources(ctx, &rs.ResourceInfo, rs.Namespace)
+		}
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	return utilerrors.NewAggregate(errs)
```

Comment:
ditto

Reply:
they output different strings (`\n`, vs `[]`) which is not great for logs but I am open to seeing other options but given these are already in our code I'd not couple the discussion with this PR